webui/react/src/services/api.ts

```diff
@@ -15,7 +15,28 @@ import {
 import { serverAddress } from 'utils/routes';
 import { isExperimentTask } from 'utils/task';
 
+/* eslint-disable @typescript-eslint/no-var-requires */
+const ndjsonStream = require('can-ndjson-stream');
+
 export const sApi = new DetSwagger.DeterminedApi(undefined, serverAddress());
+export const fetchParamCreator = DetSwagger.DeterminedApiFetchParamCreator();
+
+// FIXME why aren't the types accessible here?
+// DetSwagger.V1TrialLogsResponse;
+
+export const consumeStream = async <T = unknown>(
+  fetchArgs: DetSwagger.FetchArgs,
+  onEvent: (event: T) => void,
+  onFinish: () => void): Promise<void> => {
```

Comment:
any suggestions on the API here? @hkang1 . how does onEvent and onFinish sound

Reply:
`onEvent` and `onFinish` sound great.
questions:
1) can we capture errors through something like `onError`? I guess it depends on the streaming behavior. So for example for logs, if certain log event streams failed, will the backend try to resend the failure?
2) how does `onFinsh` get triggered? through an abrupt disconnection or possibly a user triggered disconnect?

Reply:
> will the backend try to resend the failure?

I don't think so

> how does onFinsh get triggered?

when the server successfully closes the connection. I'll test this.

Reply:
about onError how about we rely on exception handling? I added  onFinish and onEvent since there is not built-in language construct for it as there is for errors. I also wrapped the exception with errorHandler in a recent commit.